sql/lib/text-utils.js

```diff
@@ -0,0 +1,68 @@
+const writingScriptDetector = (() => {
```

Comment:
I named this `text-utils` because I can see myself adding a couple of other methods if I get around to writing the other optional queries. Happy to rename if it is too generic.